src/test/java/io/sniffy/socket/SnifferSocketImplTest.java

```diff
@@ -0,0 +1,149 @@
+package io.sniffy.socket;
+
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.mockito.Mock;
+import org.powermock.core.classloader.annotations.PrepareForTest;
+import org.powermock.modules.junit4.PowerMockRunner;
+
+import java.io.FileDescriptor;
+import java.io.IOException;
+import java.net.InetAddress;
+import java.net.SocketImpl;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.fail;
+import static org.mockito.Matchers.anyInt;
+import static org.powermock.api.mockito.PowerMockito.*;
+
+@RunWith(PowerMockRunner.class)
+@PrepareForTest(SocketImpl.class)
+public class SnifferSocketImplTest {
+
+    @Mock
+    private SocketImpl delegate;
+
+    private SnifferSocketImpl sniffySocket;
+
+    @Before
+    public void createSniffySocket() {
+        sniffySocket = new SnifferSocketImpl(delegate);
+    }
+
+    @Test
+    public void testSendUrgentData() throws Exception {
+
+        sniffySocket.sendUrgentData(1);
+
+        verifyPrivate(delegate).invoke("sendUrgentData",1);
+        verifyNoMoreInteractions(delegate);
+
+    }
+
+    @Test
+    public void testSendUrgentDataThrowIOException() throws Exception {
+
+        IOException expected = new IOException();
+        when(delegate, "sendUrgentData", anyInt()).thenThrow(expected);
+
+        try {
+            sniffySocket.sendUrgentData(1);
+            fail();
+        } catch (IOException actual) {
+            assertEquals(expected, actual);
+        }
+
+        verifyPrivate(delegate).invoke("sendUrgentData",1);
+        verifyNoMoreInteractions(delegate);
+
+    }
+
+    @Test
+    public void testSendUrgentDataThrowRuntimeException() throws Exception {
+
+        RuntimeException expected = new RuntimeException();
+        when(delegate, "sendUrgentData", anyInt()).thenThrow(expected);
+
+        try {
+            sniffySocket.sendUrgentData(1);
+            fail();
+        } catch (Exception actual) {
+            assertEquals(expected, actual);
+        }
+
+        verifyPrivate(delegate).invoke("sendUrgentData",1);
+        verifyNoMoreInteractions(delegate);
+
+    }
+
+    @Test
+    public void testShutdownInput() throws Exception {
+
+        sniffySocket.shutdownInput();
+
+        verifyPrivate(delegate).invoke("shutdownInput");
+        verifyNoMoreInteractions(delegate);
+
+    }
+
+    @Test
+    public void testShutdownOutput() throws Exception {
```

Comment:
![Codacy](https://www.codacy.com/assets/images/favicon.png) Issue found: [JUnit tests should include assert() or fail()](https://www.codacy.com/app/sniffy/sniffy/file/2232257618/issues/source?bid=3005960&fileBranchId=3034483#l91)